src/bin/main.rs

```diff
@@ -11,21 +7,6 @@ async fn main() -> anyhow::Result<()> {
     }
 
     env_logger::init();
-
-    log::info!("Serving on http://0.0.0.0:7878");
-
-    let state = State::new().context("Failed to initialze \"State\"!")?;
-    let state = Data::new(state);
-
-    HttpServer::new(move || {
-        // `Data<State>` is being cloned to avoid creating
-        // one `State` instance per worker.
-        // Otherwise, `Data<State>` will be intialized as many
-        // times as the total count of workers configured
-        App::new().app_data(state.clone()).configure(router)
-    })
-    .bind("0.0.0.0:7878")?
-    .run()
-    .await?;
+    run("0.0.0.0:7878").await?;
```

Comment:
What do you think about:

1. Take HOST or use a default from environment variables
2. Take PORT or use a default from environment variables
3. Create an instance of `std::net::SocketAddr` with provided or default HOST/PORT environment variables

Thing is that I think its better to rely on types than primitives for values that require validation. Using a `SocketAddr` here is more semantic and also give us validation OOTB for this use case

Reply:
Yes, I agree on that, but in any case I think that it could be done in another PR, we can create an Issue for that

Reply:
SGTM!